### PR TITLE
fix: Add aria attributes to facets

### LINKED ISSUE #11637: Collapsible elements need aria-controls and aria-expanded [work on PR #14880/Windows]
Product tabs in Product Details Page and Product facets in Product Listing Page need aria attributes such as aria-controls and aria-expanded.

Probably a service and directive will be helpful to track state of aria-expanded

### PR DESCRIPTION
**Pull request by Zeyber**
PLP Part of #11637

### CODE REVIEW

**Comment by developpeurweb** [comment]
**Test outcome for PLP:**
- **JAWS** - Brakes pattern. Reads whole content of the panel, then button expanded state at the end, ENTER key gets overridden and will not work to enter into the panel,  it only works for expanding button. 
- **NVDA** -  Forces accessible pattern, so it's mostly fine.
- **VO** - Follows implemented custom pattern, which never lands on the button, but on the first item.

**Some thoughts/options**

- Removing the double-enter and double-escape pattern.
- Aux key events (up/down arrow to jump groups, TAB to go one-by-one)

**Also:**
- Let's try IDs without spaces and associating the buttons.
- Do we need IDs? if we remove aria-controls and use aria-label, will it sound the same?

Screen Shot 2022-01-31 at 5 51 48 PM
Screen Shot 2022-01-31 at 5 55 34 PM
Screen Shot 2022-01-31 at 5 58 55 PM

**Comment by developpeurweb** [comment]
- We still have some issues with `aria-labelledby` linking to IDs.
- Is it possible to remove the double-enter and double-escape pattern? 
It would be a better cross-screen-reader experience if the user just follows the source order, item-by-item when they are available once a given panel is open. The facet "group skipping" has different behaviour across SRs.

Screen Shot 2022-02-01 at 3 43 51 PM
Screen Shot 2022-02-01 at 3 44 20 PM

**Review by developpeurweb — DISMISSED** [review]
This works well on JAWS/NVDA/VO

File: docs/migration/5_0.md [hunk 1]
```diff
@@ -359,6 +359,11 @@ The following keys have been added to `configurator-common.ts`:
 
 - `store-finder-list-item.component.html` markup structure changed. `h2.cx-store-name > button` changed to `a.cx-store-name` for screen reader improvements.
 
+### FacetListComponent
```
Comment (Matejk00) [inline docs/migration/5_0.md]: It is doubled section for `### FacetListComponent` in 5_0 document. We can add this info to existing one under "Template Changes".

**Review by Matejk00 — CHANGES_REQUESTED** [review]

**Review by Matejk00 — APPROVED** [review]

**Review by developpeurweb — APPROVED** [review]
This works well with JAWS/NVDA/VO

**Pull request merged**